Sources/StreamChatSwiftUI/CommonViews/SearchBar.swift

```diff
@@ -0,0 +1,73 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import SwiftUI
+
+struct SearchBar: View {
+    
+    @Injected(\.colors) private var colors
+    @Injected(\.fonts) private var fonts
+    @Injected(\.images) private var images
+    
+    @Binding var text: String
+    @State private var isEditing = false
+        
+    var body: some View {
+        HStack {
+            TextField(L10n.Message.Search.title, text: $text)
+                .padding(8)
+                .padding(.leading, 8)
+                .padding(.horizontal, 24)
+                .background(Color(colors.background1))
+                .cornerRadius(18)
+                .overlay(
+                    HStack {
+                        Image(uiImage: images.searchIcon)
+                            .customizable()
+                            .aspectRatio(contentMode: .fit)
+                            .foregroundColor(Color(colors.textLowEmphasis))
+                            .frame(maxHeight: 18)
+                            .padding(.leading, 12)
+                        
+                        Spacer()
+                        
+                        if isEditing {
+                            Button(action: {
+                                self.text = ""
```

Comment:
When the searchfield is empty this will do nothing. It intuitively felt to me that it should stop the editing in this case. Not sure if this is expected behavior, though, as it would mimic the behavior of the cancel button.

Reply:
I'll try to hide it in this case.